load-testing/tests/relays_stress_test.go

```diff
@@ -0,0 +1,457 @@
+		func(ctx context.Context, notif *stakingInfoNotif) (*stakingInfoNotif, bool) {
+			// Ensure that new gateways and suppliers are staked.
+			// Ensure that new applications are funded and have an account entry on-chain
+			// so that they can stake and delegate in the next block.
+			txResults = s.waitForTxsToBeCommitted()
+			s.ensureFundedActors(txResults, notif.newApps)
+			s.ensureStakedActors(txResults, MsgStakeGateway, notif.newGateways)
+			s.ensureStakedActors(txResults, MsgStakeSupplier, notif.newSuppliers)
+
+			// Update the list of staked suppliers.
+			s.stakedSuppliers = append(s.stakedSuppliers, notif.newSuppliers...)
+
+			// If no apps or gateways are to be staked, skip the rest of the process.
+			if len(notif.newApps) == 0 && len(notif.newGateways) == 0 {
+				return nil, true
+			}
+
+			s.sendStakeAndDelegateAppsTxs(&notif.sessionInfoNotif, notif.newApps, notif.newGateways)
+
+			return notif, false
+		},
+	)
+
+	// When staking and delegation transactions are sent, wait for them to be committed
+	// before adding the new actors to the list of prepared actors to be activated in
+	// the next session.
+	channel.ForEach(s.ctx, stakedAndDelegatingObs,
+		func(ctx context.Context, notif *stakingInfoNotif) {
+			// Wait for the next block to commit staking and delegation transactions
+			// and be able to send relay requests evenly distributed across all gateways.
+			txResults = s.waitForTxsToBeCommitted()
+			s.ensureStakedActors(txResults, MsgStakeApplication, notif.newApps)
+			s.ensureDelegatedApps(txResults, s.activeApplications, notif.newGateways)
+			s.ensureDelegatedApps(txResults, notif.newApps, notif.newGateways)
+			s.ensureDelegatedApps(txResults, notif.newApps, s.activeGateways)
+
+			// Add the new actors to the list of prepared actors to be activated in
+			// the next session.
+			s.preparedApplications = append(s.preparedApplications, notif.newApps...)
+			s.preparedGateways = append(s.preparedGateways, notif.newGateways...)
+		},
+	)
+}
+
+func (s *relaysSuite) ALoadOfConcurrentRelayRequestsAreSentFromTheApplications() {
+	// Limit the number of concurrent requests to maxConcurrentRequestLimit.
+	batchLimiter := sync2.NewLimiter(maxConcurrentRequestLimit)
+
+	channel.ForEach(s.ctx, s.batchInfoObs, s.sendRelayBatchFn(batchLimiter))
+
+	// Block the feature step until the test is done.
+	<-s.ctx.Done()
+}
+
+func (s *relaysSuite) PairsOfClaimAndProofMessagesShouldBeCommittedOnchain(a string) {
+	txEventsCtx, cancelTxEvents := context.WithCancel(s.ctx)
+	_ = cancelTxEvents
+
+	txEventsCh := s.newTxEventsObs.Subscribe(txEventsCtx).Ch()
+	for txEvent := range txEventsCh {
+		// TODO_IMPROVE: refactor this and other similar loops over events.
+		for _, event := range txEvent.Result.Events {
+			if event.Type != "message" {
+				continue
+			}
+
+			if hasEventAttr(event.Attributes, "action", MsgCreateClaim) {
+				// TODO_TECHDEBT: match up claims & proofs and assert their quantity.
```

Comment:
Can you create a ticket for this so we can tend to it in the next iteration (or later this week)? Don't want it to block the PR.

Reply:
The work for this has already been done on another branch. PR forthcoming. Do you still think we need an issue for it?